src/views/CreateTender/CreateTender.vue

```diff
@@ -0,0 +1,361 @@
+        tenderTypes: [],
+        name: null,
+        nationalRegNumber: null,
+        countryId: null,
+        townCity: null,
+        contactPersonName: null,
+        contactPersonSurname: null,
+        phoneNumber: null,
+        cvpCodeId: null,
+        tenderTypeId: null,
+        description: null,
+        minValue: null,
+        maxValue: null,
+        currencyId: null,
+        publicDate: null,
+        deadLineOfSub: null,
+        deadForSinging: null,
+
+        contractFile: null,
+        fileKeys: new Map(),
+        declineFile: null,
+        declineFileKey: null,
+        awardFile: null,
+        awardFileKey: null,
+      }
+    },
+
+    methods: {
+      uploadContractFile() {
+        this.contractFile = this.$refs.contractFile.files[0];
+      },
+      uploadDeclineFile() {
+        this.declineFile = this.$refs.declineFile.files[0];
+      },
+      uploadAwardFile() {
+        this.awardFile = this.$refs.awardFile.files[0];
+      },
+      submitFile(file, fileKeyKey) {
+        const formData = new FormData();
+        formData.append('file', file);
+        const headers = {'Content-Type': 'multipart/form-data'};
+        fetch('http://localhost:8080/uploadFile', {
+          method: 'POST',
+          headers: {}
+              .then((res) => {
+                if (res.status === 200) {
+                  this.fileKeys.set(fileKeyKey, res.data)
+                }
+                this.contractFileKey = res.data;
+              })
+        })
+      },
+      publishTender() {
+        fetch(API_BASE_URL+'/tenders', {
+          method: 'POST',
+          "name": this.name,
+          "nationalRegNumber": this.nationalRegNumber,
+          "countryId": this.countryId,
+          "townCity": this.townCity,
+          "contactPersonName": this.contactPersonName,
+          "contactPersonSurname": this.contactPersonSurname,
+          "phoneNumber": this.phoneNumber,
+          "cvpCodeId": this.cvpCodeId,
+          "tenderTypeId": this.tenderTypeId,
+          "description": this.description,
+          "minValue": this.minValue,
+          "maxValue": this.maxValue,
+          "currencyId": this.currencyId,
+          "contractFileKey": this.fileKeys.get('contractFileKey'),
+          "awardFileKey": this.fileKeys.get('awardFileKey'),
+          "declineFileKey": this.fileKeys.get('declineFileKey'),
+          "publicDate": this.publicDate ? this.publicDate.getTime() : null,
+          "deadLineOfSub": this.deadLineOfSub ? new Date(this.deadLineOfSub).getTime() : null,
+          "deadForSinging": this.deadForSinging ? new Date(this.deadForSinging).getTime() : null,
+
+          headers: getHeaders(),
+        })
+
+                  .then(response => {
+                    this.$emit('created', true)
+                  })
+        }
+      },
+      cancelTender(){
+        this.$emit('created', false)
+      },
+
+  }
+
+</script>
+
+<style scoped>
+@import "@/styles/main.css";
+.publish-button {
```

Comment:
publish-button and cancel-button are almost the same. create a generic button class called _button_ and put only styles that diverge in _publish-button_ and _cancel-button_ . Also, check out about BEM methodology